packages/hashtower.circom/circuits/hashtower-hash-chain.circom

```diff
@@ -0,0 +1,228 @@
+pragma circom 2.1.4;
+
+include "circomlib/circuits/multiplexer.circom";
+include "circomlib/circuits/poseidon.circom";
+include "circomlib/circuits/comparators.circom";
+include "circomlib/circuits/gates.circom";
+
+// Pick in[sel]
+template PickOne(N) {
+    signal input in[N];
+    signal input sel;
+    signal output out;
+
+    component mux = Multiplexer(1, N);
```

Comment:
I think a cheaper approach here would be

```
signal vals[N];
var _out = 0;
for (var x = 0; x < N; x++) {
  vals[x] <== IsEqual()([x, sel]) * in[x];
  _out += vals[x];
}
out <== _out;
```

Reply:
Some experiments:
https://gist.github.com/LCamel/e3dd35efa52338d89c13d913c6be5d42